pkg/replica/revision_counter.go

```diff
@@ -36,14 +36,21 @@ func (r *Replica) readRevisionCounter() (int64, error) {
 	return counter, nil
 }
 
+var revisionCounterBuf = sparse.AllocateAligned(revisionBlockSize)
+
 func (r *Replica) writeRevisionCounter(counter int64) error {
 	if r.revisionFile == nil {
 		return fmt.Errorf("BUG: revision file wasn't initialized")
 	}
 
-	buf := make([]byte, revisionBlockSize)
-	copy(buf, []byte(strconv.FormatInt(counter, 10)))
-	_, err := r.revisionFile.WriteAt(buf, 0)
+	counterBytes := []byte(strconv.FormatInt(counter, 10))
+	copy(revisionCounterBuf, counterBytes)
+	// Need to clear the part of the slice left over from the previous function call.
+	// The maximum length of the string representation of an int64 is 20 bytes so only need to clear upto 20 bytes
```

Comment:
We discussed this yesterday. I was of the opinion that we should maintain two versions of this function:

- One that allocates its own buffer to be called generically by anyone. (This is to ensure safety in case a generic caller might decrement the revisionCounter.)
- One that uses the pre-allocated buffer to be called only by increaseRevisionCounter. (I think the clearing logic here is not necessary for that one.)

When we were discussing it, @Phan mentioned that the code is just as fast with this clearing logic as it is without, so it seems a reasonable safety measure to leave it in for all callers.

Reply:
Thank you for the recall! Yes, it doesn't seem to slow down so I think it is fine to have the cleaning logic in all callers unless we have additional concerns.